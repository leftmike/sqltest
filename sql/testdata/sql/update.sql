--
-- Test UPDATE
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (c1 int primary key, c2 int, c3 int);

INSERT INTO tbl1 VALUES
    (0, 1, 2),
    (3, 4, 5),
    (6, 7, 8),
    (9, 10, 11),
    (12, 13, 14),
    (15, 16, 17),
    (18, 19, 20),
    (21, 22, 23),
    (24, 25, 26),
    (27, 28, 29),
    (30, 31, 32);

SELECT * FROM tbl1;

UPDATE tbl1 SET c2 = c1 + c3 WHERE c1 < 6;

SELECT * FROM tbl1;

UPDATE tbl1 SET c3 = c3 * 5 WHERE c2 % 2 = 0;

SELECT * FROM tbl1;
