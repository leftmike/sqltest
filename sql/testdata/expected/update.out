--
-- Test UPDATE
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (c1 int primary key, c2 int, c3 int);
INSERT INTO tbl1 VALUES
    (0, 1, 2),
    (3, 4, 5),
    (6, 7, 8),
    (9, 10, 11),
    (12, 13, 14),
    (15, 16, 17),
    (18, 19, 20),
    (21, 22, 23),
    (24, 25, 26),
    (27, 28, 29),
    (30, 31, 32);
SELECT * FROM tbl1;
    c1 c2 c3
    -- -- --
  1  0  1  2
  2 12 13 14
  3 15 16 17
  4 18 19 20
  5 21 22 23
  6 24 25 26
  7 27 28 29
  8  3  4  5
  9 30 31 32
 10  6  7  8
 11  9 10 11
(11 rows)
UPDATE tbl1 SET c2 = c1 + c3 WHERE c1 < 6;
SELECT * FROM tbl1;
    c1 c2 c3
    -- -- --
  1  0  2  2
  2 12 13 14
  3 15 16 17
  4 18 19 20
  5 21 22 23
  6 24 25 26
  7 27 28 29
  8  3  8  5
  9 30 31 32
 10  6  7  8
 11  9 10 11
(11 rows)
UPDATE tbl1 SET c3 = c3 * 5 WHERE c2 % 2 = 0;
SELECT * FROM tbl1;
    c1 c2  c3
    -- --  --
  1  0  2  10
  2 12 13  14
  3 15 16  85
  4 18 19  20
  5 21 22 115
  6 24 25  26
  7 27 28 145
  8  3  8  25
  9 30 31  32
 10  6  7   8
 11  9 10  55
(11 rows)
